README.md

```diff
@@ -804,22 +804,26 @@ Scenario has secrets. Here's how you use them.
 state = scenario.State(
     secrets={
         scenario.Secret(
-            {0: {'key': 'public'}},
-            id='foo',
-        ),
-    },
+            current={'key': 'public'},
```

Comment:
wondering if this should be 'content' instead, or `current_content`.
Also should it be positional?

Reply:
or 'payload', as you call it below. What is the official terminology?

Reply:
> or 'payload', as you call it below. What is the official terminology?

The term I see most is "value" (because of the key/value nature, I assume), but I see "payload" sometimes. "content" matches the `get_content`, `set_content` used in ops. I can check with Ian or Dora to find out what the official term is, if there is one.

I think maybe "current" should be "tracked"?

My preference is that you don't have to specify both latest and current/tracked, because there seem to be a bunch of common test cases where you only care about one revision. I'd like them to both be positional, but I also dislike APIs where I can't remember which argument goes first (`ls`, most of Python's `re`, etc). It felt like providing "latest" and having that copied to "current" made most sense, but if they're both positional, then the natural order seems like "current, latest" not "latest, current". Alternatively, they could be a single attribute of the Secret, but then it's more work to make one.

Some variations:

```
secret = Secret(latest, *, tracked=None, id=..., owner=None, ...)
secret.tracked
secret.latest
```

```
secret = Secret(tracked, latest=None, *, id=..., owner=None, ...)
secret.tracked
secret.latest
```

```
secret = Secret({"tracked": tracked, "latest": latest}, *, id=..., owner=None, ...)
secret.content["tracked"]
secret.content["latest"]
```

```
secret = Secret(SecretContent(tracked, latest), *, id=..., owner=None, ...)
secret.content.tracked
secret.content.latest
```

I don't really like having a new class just to be able to dot-access, and I don't love dicts of dicts. So I think I like the second one above the most.

Reply:
Ian says "content" is the term. I guess it could be:

```python
secret = Secret(tracked_content, latest_content=None, *, id=..., owner=None, ...)
secret.tracked_content
secret.latest_content
```

It's a bit longer, although if when constructing people generally use positional arguments for the content then that doesn't matter there, it would only be in the various asserts afterwards. It is clearer - particularly if you don't see the type then `tracked` or `latest` could easily be Boolean values, but `tracked_content` is clearly going to be the secret content.

@benhoyt any preference on naming here?

Reply:
Yeah, I think "content" in the name is clearer than without, and I like the "tracked" and "latest" prefixes.

I'm not sure whether `latest_content` should be keyword-only -- I'm thinking probably so.

Reply:
> I'm not sure whether `latest_content` should be keyword-only -- I'm thinking probably so.

Because it's not obvious whether it's tracked,latest or latest,tracked - or for some other reason? If it's switched so that latest is copied from tracked if missing (rather than the other way, as the branch has now) then I prefer allowing both as positional - I think oldest to newest going left to right is predictable enough.

Reply:
> Because it's not obvious whether it's tracked,latest or latest,tracked - or for some other reason?

Yeah, because the order is not obvious.

> I prefer allowing both as positional

Fine with me. I'm not so sure oldest to newest is predictable, but I don't have strong feelings about it.